src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -104,22 +102,41 @@ private bool TryRewriteTagHelper(Block tagBlock, RewritingContext context)
 
             var descriptors = Enumerable.Empty<TagHelperDescriptor>();
 
-            if (IsPotentialTagHelper(tagName, tagBlock))
-            {
-                descriptors = _provider.GetTagHelpers(tagName);
-            }
-
-            // If there aren't any TagHelperDescriptors registered then we aren't a TagHelper
-            if (!descriptors.Any())
+            if (!IsPotentialTagHelper(tagName, tagBlock))
             {
                 return false;
             }
 
+            var currentTagBlockTracker = _tagStack.Count > 0 ? _tagStack.Peek() : null;
+            var tagNameScope = currentTagBlockTracker?.Builder.TagName ?? string.Empty;
+
             if (!IsEndTag(tagBlock))
             {
-                // We're in a begin tag helper block
+                // We're in a begin tag block
+
```

Comment:
then put it above line 113 and call out fact you're describing the `true` case.  as it is, comment still looks like it describes the immediately-following code and made me believe line 116 was useless whitespace.